crates/bevy_asset/src/asset_server.rs

```diff
@@ -455,6 +531,7 @@ impl AssetServer {
         }
     }
 
+    /// Iterates through asset references and marks assets with no active handles as unused.
```

Comment:
Why is this (and free_unused_assets) necessary? Does the user need to call these to clean up unused assets? I thought cleanup was automatic using RAII on strong handles?
I'm guessing that it's so the user can control when assets are freed by making their own alternative to the free_unused_assets_system, but having it in docs would make it clearer.

Reply:
Well, it's an implementation detail of the asset plugin, I'm not sure why they are public functions either. Handle usage does use RAII but it is event-driven, so the `free_unused_assets_system` is added to the `PreUpdate` stage (see `lib.rs`) to look through these events and update reference counts. They're freed in `update_asset_storage`.

If I were to add documentation about controlling asset lifetime by yourself, it could be as a section in the `AssetServer` struct docstring, but such a custom system would conflict with the builtin cleanup system as you can't remove it (at the moment), so this is out of scope for this PR. What I could do is to make them private, as it isn't used anywhere else. What do you think @alice-i-cecile?

Reply:
I think we leave this as pub for now to reduce controversy, and then tackle their pub nature in a followup PR.